src/Microsoft.DotNet.Build.Tasks/ParseTestCoverageInfo.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Xml;
+using System.Collections.Generic;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ParseTestCoverageInfo : Task
+    {
+        // Path to the directory that contais *.coverage.xml coverage info.
+        [Required]
+        public string CoverageInfoDirectory { get; set; }
+
+        // One per xml parsed, key = test name, value list of modules visited
+        private Dictionary<string, Dictionary<string, ModuleInfo>> testCoverageInfo = null;
+
+        public override bool Execute()
+        {
+            if (Directory.Exists(CoverageInfoDirectory))
+            {
+                Log.LogMessage(MessageImportance.Normal, "Parsing coverage files...");
+                if (!ProcessDirectory(CoverageInfoDirectory))
+                {
+                    Log.LogError("There was no coverage file found.");
+                    return false;
+                }
+            }
+            else
+            {
+                Log.LogError("{0} is not a valid path", CoverageInfoDirectory);
+                return false;
+            }
+
+            SaveToFile();
+
+            return true;
+        }
+
+        /// <summary>
+        /// Explore the input path recursively.
+        /// Parse opencoverage output.
+        /// </summary>
+        /// <param name="directory"></param>
+        /// <returns>True if at least one file that matched the pattern coverage_[name of test assembly].xml is found</returns>
+        private bool ProcessDirectory(string directory)
+        {
+
+            string[] results = Directory.GetFiles(directory, "*.coverage.xml");
+
+            bool found = (results.Length > 0);
+
+            foreach (string file in results)
+            {
+                if (testCoverageInfo == null)
+                {
+                    testCoverageInfo = new Dictionary<string, Dictionary<string, ModuleInfo>>();
+                }
+
+                ParseCoverageFile(file);
+            }
+
+            foreach (string child in Directory.GetDirectories(directory))
+            {
+                found |= ProcessDirectory(child);
+            }
+
+            return found;
+        }
+
+        /// <summary>
+        /// Parse all coverage .xml files
+        /// </summary>
+        private void ParseCoverageFile(string file)
+        {
+            XmlDocument coverageXml = new XmlDocument();
+            coverageXml.Load(file);
```

Comment:
I'll considering the future if we get to a point were this task becomes an issue.